olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -190,13 +227,97 @@ class AuthenticationClient::Impl final {
       const AuthenticationCredentials& credentials,
       const std::string& userAccessToken, const SignOutUserCallback& callback);
 
+  client::CancellationToken IntrospectApp(std::string access_token,
+                                          IntrospectAppCallback callback);
+
   void SetNetworkProxySettings(
       const http::NetworkProxySettings& proxy_settings);
 
   void SetNetwork(std::shared_ptr<http::Network> network);
 
   void SetTaskScheduler(std::shared_ptr<TaskScheduler> task_scheduler);
 
+  static olp::authentication::IntrospectAppResult parseIntrospectAppResult(
+      const std::shared_ptr<rapidjson::Document>& doc) {
+    olp::authentication::IntrospectAppResult result;
+    if (doc->HasMember(Constants::CLIENT_ID)) {
+      result.SetClientId((*doc)[Constants::CLIENT_ID].GetString());
+    }
+    if (doc->HasMember(Constants::NAME)) {
+      result.SetName((*doc)[Constants::NAME].GetString());
+    }
+    if (doc->HasMember(Constants::DESCRIPTION)) {
+      result.SetDescription((*doc)[Constants::DESCRIPTION].GetString());
+    }
+    if (doc->HasMember(Constants::REDIRECT_URIS)) {
+      auto uris = (*doc)[Constants::REDIRECT_URIS].GetArray();
+      std::vector<std::string> value_array;
+      value_array.reserve(uris.Size());
+      for (auto& value : uris) value_array.push_back(value.GetString());
+      result.SetReditectUris(std::move(value_array));
+    }
+    if (doc->HasMember(Constants::ALLOWED_SCOPES)) {
+      auto scopes = (*doc)[Constants::ALLOWED_SCOPES].GetArray();
+      std::vector<std::string> value_array;
+      value_array.reserve(scopes.Size());
+      for (auto& value : scopes) value_array.push_back(value.GetString());
+      result.SetAllowedScopes(std::move(value_array));
+    }
+    if (doc->HasMember(Constants::TOKEN_ENDPOINT_AUTH_METHOD)) {
+      result.SetTokenEndpointAuthMethod(
+          (*doc)[Constants::TOKEN_ENDPOINT_AUTH_METHOD].GetString());
+    } else {
+      result.SetTokenEndpointAuthMethod(
+          Constants::DEFAULT_TOKEN_ENDPOINT_AUTH_METHOD);
```

Comment:
Why do we need to set a default one?

Reply:
REST API says `The default value will be "client_secret_jwt" if the field is not included in the request body`, so an empty auth method field actually means `client_secret_jwt`.
Is this user responsibility to handle such case, so we can left it empty?

Reply:
This `The default value will be "client_secret_jwt" if the field is not included in the request body` doesn't mean that you will not receive this value. Or?
So this all depends on the type of token passed by the user. I would not set any default values. Let the backend set what he thinks is necessary.